src/Features/CSharp/Portable/RemoveUnnecessaryImports/AbstractCSharpRemoveUnnecessaryImportsService.cs

```diff
@@ -35,9 +35,9 @@ internal partial class AbstractCSharpRemoveUnnecessaryImportsService :
                 }
 
                 var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
+                
                 var oldRoot = (CompilationUnitSyntax)root;
-                var newRoot = (CompilationUnitSyntax)new Rewriter(unnecessaryImports, cancellationToken).Visit(oldRoot);
+                var newRoot = (CompilationUnitSyntax)new Rewriter(unnecessaryImports, this, document, cancellationToken).Visit(oldRoot);
```

Comment:
@CyrusNajmabadi is this what you meant by `pass along the AbstractRemoveUnnecessaryImportsService which the rewriter can store` ?